contents/data/nosql/NoSQL.md

```diff
@@ -0,0 +1,430 @@
+in NoSQL, this means that developers can adapt the database quicker to changing customer and business requirements.
+
+
+### 2. Horizontal scaling :fas-server:
+
+Another key benefit of NoSQL databases is the ability to scale horizontally ( distributing workload across
+multiple servers ), without discarding much of its key features.
+
+This is largely due to the schema-less architecture of such databases, allowing data to split across
+multiple servers easily and efficiently.
+
+For example, take the following collection of items with a `title`:
+
+```js
+[
+  {
+    title: "let's assume item titles are lexicographically distributed uniformly over the long run"
+    type: "book",
+    price: 20,
+    popularity: 9.7,
+  },
+  // item 2
+  // item 3
+  // ....
+  // item N
+]
+```
+
+Assuming we don't have relations from items to themselves inside these documents,
+we can split the collection like so:
+
+<img src="horizontalScalability.png" width="500" />
+
+As a result, the database access workload can be distributed evenly and efficiently across
+multiple servers easily.
+
+<box type="success" light icon-size="2x">
+
+As businesses grow and see increasing amounts of web traffic, it is <em>crucial</em> that its databases can scale to meet
+consumer and business demands.
+
+<div>
+Vertical scaling ( increasing the processing power of the machine ) can only go so far until the single machine hits its limit.</div>
+</box>
+
+### 3. Widespread adoption
+
+<br>
+<figure>
+<img src="databaseSurvey.png" width="90%" style="margin:0 auto" />
+<figcaption>
+<small>Most popular database technologies from Stack Overflow developer survey 2019</small>
+</figcaption>
+</figure>
+<br>
+
+While certainly trending behind SQL databases, NoSQL databases have been booming over the past couple of years,
+due to the increasing applicability of its benefits to requirements today. ( Amazon uses a proprietary NoSQL database! )
+
+This bodes well for the maturity and development of this evolving technology, and your potential use cases for it.
+
+---
+
+## Caveats of NoSQL
+
+<br>
+
+#### 1. Lack of standardisation
+From both a user and implementation standpoint, NoSQL databases vary from one solution to another greatly, which can incur
+extra development costs in projects when there is a need to migrate to another solution, or when new developers are introduced
+to the project.
+
+This is in stark contrast to SQL, which's syntax is mostly standardised across its different flavours.
+
+#### 2. Not suited for complex relational queries
+While NoSQL databases certainly allow for more flexibility in structuring out relations, most complex queries ( eg. joins for many-many relations ) _usually_ involve structured data that can be easily represented in tabular formats. 
+
+In such instances, queries are often more performant in SQL equivalents.
+
+
+---
+
+## How to get started with NoSQL?
```

Comment:
This section seems too long. I seriously doubt someone new to this topic will have the patience to read all this. They'll probably exit this chapter through the first link in this section. See if you can shorten it to about half the current length.